# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix

# build
make

# install
install -d $DESTDIR/bin
install tinyxxd $DESTDIR/bin/xxd

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/tinyxxd-build.pc << EOF
Name: tinyxxd
Version: 1.1.0
Description: Standalone version of the hex dump utility from Vim
Requires: $REQUIRES

devel=\\
/bin/xxd
EOF

exit
--- tinyxxd-1.1.0.orig/Configure
+++ tinyxxd-1.1.0/Configure
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX ?=/s#=.*#= $PREFIX#" Makefile
--- tinyxxd-1.1.0.orig/Makefile
+++ tinyxxd-1.1.0/Makefile
@@ -1,7 +1,7 @@
 .PHONY: clean install uninstall
 
 CFLAGS ?= -std=c11 -O2 -pipe -fPIC -fno-plt -fstack-protector-strong -D_GNU_SOURCE -s -z norelro -Wall -Wextra -Wpedantic -Wfatal-errors
-PREFIX ?= /usr
+PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 DESTDIR ?=
 
